Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
!pip install rpy2
%load_ext rpy2.ipython
------------------

----- stdout -----
Requirement already satisfied: rpy2 in /opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages (3.5.17)
Requirement already satisfied: cffi>=1.15.1 in /opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages (from rpy2) (1.17.1)
Requirement already satisfied: jinja2 in /opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages (from rpy2) (3.1.6)
Requirement already satisfied: tzlocal in /opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages (from rpy2) (5.3.1)
Requirement already satisfied: pycparser in /opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages (from cffi>=1.15.1->rpy2) (2.22)
Requirement already satisfied: MarkupSafe>=2.0 in /opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages (from jinja2->rpy2) (3.0.2)
----- stderr -----
Unable to determine R home: [Errno 2] No such file or directory: 'R'
------------------

[31m---------------------------------------------------------------------------[39m
[31mValueError[39m                                Traceback (most recent call last)
[36mCell[39m[36m [39m[32mIn[1][39m[32m, line 2[39m
[32m      1[39m get_ipython().system([33m'[39m[33mpip install rpy2[39m[33m'[39m)
[32m----> [39m[32m2[39m [43mget_ipython[49m[43m([49m[43m)[49m[43m.[49m[43mrun_line_magic[49m[43m([49m[33;43m'[39;49m[33;43mload_ext[39;49m[33;43m'[39;49m[43m,[49m[43m [49m[33;43m'[39;49m[33;43mrpy2.ipython[39;49m[33;43m'[39;49m[43m)[49m

[36mFile [39m[32m/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/IPython/core/interactiveshell.py:2481[39m, in [36mInteractiveShell.run_line_magic[39m[34m(self, magic_name, line, _stack_depth)[39m
[32m   2479[39m     kwargs[[33m'[39m[33mlocal_ns[39m[33m'[39m] = [38;5;28mself[39m.get_local_scope(stack_depth)
[32m   2480[39m [38;5;28;01mwith[39;00m [38;5;28mself[39m.builtin_trap:
[32m-> [39m[32m2481[39m     result = [43mfn[49m[43m([49m[43m*[49m[43margs[49m[43m,[49m[43m [49m[43m*[49m[43m*[49m[43mkwargs[49m[43m)[49m
[32m   2483[39m [38;5;66;03m# The code below prevents the output from being displayed[39;00m
[32m   2484[39m [38;5;66;03m# when using magics with decorator @output_can_be_silenced[39;00m
[32m   2485[39m [38;5;66;03m# when the last Python token in the expression is a ';'.[39;00m
[32m   2486[39m [38;5;28;01mif[39;00m [38;5;28mgetattr[39m(fn, magic.MAGIC_OUTPUT_CAN_BE_SILENCED, [38;5;28;01mFalse[39;00m):

[36mFile [39m[32m/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/IPython/core/magics/extension.py:33[39m, in [36mExtensionMagics.load_ext[39m[34m(self, module_str)[39m
[32m     31[39m [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m module_str:
[32m     32[39m     [38;5;28;01mraise[39;00m UsageError([33m'[39m[33mMissing module name.[39m[33m'[39m)
[32m---> [39m[32m33[39m res = [38;5;28;43mself[39;49m[43m.[49m[43mshell[49m[43m.[49m[43mextension_manager[49m[43m.[49m[43mload_extension[49m[43m([49m[43mmodule_str[49m[43m)[49m
[32m     35[39m [38;5;28;01mif[39;00m res == [33m'[39m[33malready loaded[39m[33m'[39m:
[32m     36[39m     [38;5;28mprint[39m([33m"[39m[33mThe [39m[38;5;132;01m%s[39;00m[33m extension is already loaded. To reload it, use:[39m[33m"[39m % module_str)

[36mFile [39m[32m/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/IPython/core/extensions.py:62[39m, in [36mExtensionManager.load_extension[39m[34m(self, module_str)[39m
[32m     55[39m [38;5;250m[39m[33;03m"""Load an IPython extension by its module name.[39;00m
[32m     56[39m 
[32m     57[39m [33;03mReturns the string "already loaded" if the extension is already loaded,[39;00m
[32m     58[39m [33;03m"no load function" if the module doesn't have a load_ipython_extension[39;00m
[32m     59[39m [33;03mfunction, or None if it succeeded.[39;00m
[32m     60[39m [33;03m"""[39;00m
[32m     61[39m [38;5;28;01mtry[39;00m:
[32m---> [39m[32m62[39m     [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[43m.[49m[43m_load_extension[49m[43m([49m[43mmodule_str[49m[43m)[49m
[32m     63[39m [38;5;28;01mexcept[39;00m [38;5;167;01mModuleNotFoundError[39;00m:
[32m     64[39m     [38;5;28;01mif[39;00m module_str [38;5;129;01min[39;00m BUILTINS_EXTS:

[36mFile [39m[32m/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/IPython/core/extensions.py:77[39m, in [36mExtensionManager._load_extension[39m[34m(self, module_str)[39m
[32m     75[39m [38;5;28;01mwith[39;00m [38;5;28mself[39m.shell.builtin_trap:
[32m     76[39m     [38;5;28;01mif[39;00m module_str [38;5;129;01mnot[39;00m [38;5;129;01min[39;00m sys.modules:
[32m---> [39m[32m77[39m         mod = [43mimport_module[49m[43m([49m[43mmodule_str[49m[43m)[49m
[32m     78[39m     mod = sys.modules[module_str]
[32m     79[39m     [38;5;28;01mif[39;00m [38;5;28mself[39m._call_load_ipython_extension(mod):

[36mFile [39m[32m/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/importlib/__init__.py:126[39m, in [36mimport_module[39m[34m(name, package)[39m
[32m    124[39m             [38;5;28;01mbreak[39;00m
[32m    125[39m         level += [32m1[39m
[32m--> [39m[32m126[39m [38;5;28;01mreturn[39;00m [43m_bootstrap[49m[43m.[49m[43m_gcd_import[49m[43m([49m[43mname[49m[43m[[49m[43mlevel[49m[43m:[49m[43m][49m[43m,[49m[43m [49m[43mpackage[49m[43m,[49m[43m [49m[43mlevel[49m[43m)[49m

[36mFile [39m[32m<frozen importlib._bootstrap>:1204[39m, in [36m_gcd_import[39m[34m(name, package, level)[39m

[36mFile [39m[32m<frozen importlib._bootstrap>:1176[39m, in [36m_find_and_load[39m[34m(name, import_)[39m

[36mFile [39m[32m<frozen importlib._bootstrap>:1147[39m, in [36m_find_and_load_unlocked[39m[34m(name, import_)[39m

[36mFile [39m[32m<frozen importlib._bootstrap>:690[39m, in [36m_load_unlocked[39m[34m(spec)[39m

[36mFile [39m[32m<frozen importlib._bootstrap_external>:940[39m, in [36mexec_module[39m[34m(self, module)[39m

[36mFile [39m[32m<frozen importlib._bootstrap>:241[39m, in [36m_call_with_frames_removed[39m[34m(f, *args, **kwds)[39m

[36mFile [39m[32m/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/rpy2/ipython/__init__.py:1[39m
[32m----> [39m[32m1[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01m.[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m rmagic
[32m      3[39m load_ipython_extension = rmagic.load_ipython_extension

[36mFile [39m[32m/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/rpy2/ipython/rmagic.py:54[39m
[32m     51[39m [38;5;28;01mimport[39;00m[38;5;250m [39m[34;01mtextwrap[39;00m
[32m     52[39m [38;5;28;01mimport[39;00m[38;5;250m [39m[34;01mtyping[39;00m
[32m---> [39m[32m54[39m [38;5;28;01mimport[39;00m[38;5;250m [39m[34;01mrpy2[39;00m[34;01m.[39;00m[34;01mrinterface_lib[39;00m[34;01m.[39;00m[34;01mcallbacks[39;00m
[32m     56[39m [38;5;28;01mimport[39;00m[38;5;250m [39m[34;01mrpy2[39;00m[34;01m.[39;00m[34;01mrinterface[39;00m[38;5;250m [39m[38;5;28;01mas[39;00m[38;5;250m [39m[34;01mri[39;00m
[32m     58[39m [38;5;28;01mimport[39;00m[38;5;250m [39m[34;01mrpy2[39;00m[34;01m.[39;00m[34;01mrinterface_lib[39;00m[34;01m.[39;00m[34;01mopenrlib[39;00m

[36mFile [39m[32m/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/rpy2/rinterface_lib/callbacks.py:11[39m
[32m      9[39m [38;5;28;01mimport[39;00m[38;5;250m [39m[34;01mtyping[39;00m
[32m     10[39m [38;5;28;01mimport[39;00m[38;5;250m [39m[34;01mos[39;00m
[32m---> [39m[32m11[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mrpy2[39;00m[34;01m.[39;00m[34;01mrinterface_lib[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m openrlib
[32m     12[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mrpy2[39;00m[34;01m.[39;00m[34;01mrinterface_lib[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m ffi_proxy
[32m     13[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mrpy2[39;00m[34;01m.[39;00m[34;01mrinterface_lib[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m conversion

[36mFile [39m[32m/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/rpy2/rinterface_lib/openrlib.py:58[39m
[32m     56[39m     rlib = _rinterface_cffi.lib
[32m     57[39m [38;5;28;01melse[39;00m:
[32m---> [39m[32m58[39m     rlib = [43m_dlopen_rlib[49m[43m([49m[43mR_HOME[49m[43m)[49m
[32m     61[39m [38;5;66;03m# R macros and functions[39;00m
[32m     62[39m [38;5;28;01mdef[39;00m[38;5;250m [39m[34m_get_symbol_or_fallback[39m(symbol: [38;5;28mstr[39m, fallback: typing.Any):

[36mFile [39m[32m/opt/hostedtoolcache/Python/3.11.11/x64/lib/python3.11/site-packages/rpy2/rinterface_lib/openrlib.py:45[39m, in [36m_dlopen_rlib[39m[34m(r_home)[39m
[32m     41[39m [38;5;250m[39m[33;03m"""Open R's shared C library.[39;00m
[32m     42[39m 
[32m     43[39m [33;03mThis is only relevant in ABI mode."""[39;00m
[32m     44[39m [38;5;28;01mif[39;00m r_home [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[32m---> [39m[32m45[39m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m([33m'[39m[33mr_home is None. [39m[33m'[39m
[32m     46[39m                      [33m'[39m[33mTry python -m rpy2.situation[39m[33m'[39m)
[32m     47[39m lib_path = rpy2.situation.get_rlib_path(r_home, platform.system())
[32m     48[39m [38;5;28;01mif[39;00m lib_path [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:

[31mValueError[39m: r_home is None. Try python -m rpy2.situation

